packages/commonwealth/server/routes/communities/put_community_stakes_handler.ts

```diff
@@ -0,0 +1,57 @@
+import { AppError } from '@hicommonwealth/adapters';
+import { Community, DB } from '@hicommonwealth/model';
+import { CommunityStakeAttributes } from '@hicommonwealth/model/build/models/community_stake';
+import { ServerControllers } from '../../routing/router';
+import { TypedRequest, TypedResponse, success } from '../../types';
+import { validateCommunityStakeConfig } from '../../util/commonProtocol/communityStakeConfigValidator';
+import { formatErrorPretty } from '../../util/errorFormat';
+
+type PutCommunityStakesParams = Community.SetCommunityStakeParams;
+type PutCommunityStakesBody = Community.SetCommunityStakeBody;
+type PutCommunityStakesResponse = CommunityStakeAttributes;
+
+export const putCommunityStakeHandler = async (
+  models: DB,
+  controllers: ServerControllers,
+  req: TypedRequest<PutCommunityStakesBody, any, PutCommunityStakesParams>,
+  res: TypedResponse<PutCommunityStakesResponse>,
+) => {
+  const paramsValidationResult =
+    Community.SetCommunityStakeParamsSchema.safeParse(req.params);
+
+  if (paramsValidationResult.success === false) {
+    throw new AppError(formatErrorPretty(paramsValidationResult));
+  }
+
+  const { community_id, stake_id } = paramsValidationResult.data;
+
+  const adminAddress = (await req.user.getAddresses())
```

Comment:
This is ok for now, but it's code duplication... we have middleware to check for admin auth